qan/analyzer/mysql/worker/perfschema/worker.go

```diff
@@ -421,23 +401,12 @@ ROW_LOOP:
 				class.Rows[row.Schema] = row
 			} else {
 				// Get class digest text (fingerprint).
-				var digestText string
-				if prevClass, havePrevClass := prev[classId]; havePrevClass {
-					// Class was in previous iter, so re-use its digest text.
-					digestText = prevClass.DigestText
-				} else {
-					// Have never seen class before, so get digext text from perf schema.
-					var err error
-					digestText, err = w.getText(row.Digest)
-					if classId == "2" && digestText == "" {
-						// To make explains works
-						digestText = `-- performance_schema.events_statements_summary_by_digest is full`
-					}
-					if err != nil {
-						w.logger.Error(err)
-						continue
-					}
+				digestText := row.DigestText
+				if classId == "2" && digestText == "" {
+					// To make explains works
+					digestText = `-- performance_schema.events_statements_summary_by_digest is full`
 				}
+
```

Comment:
I changed this part. I don't think it makes sense now to check if `prevClass` has DigestText as you can simply take it directly from `row`.